options/options_test.cc

```diff
@@ -3441,6 +3441,32 @@ TEST_F(OptionTypeInfoTest, TestStruct) {
   ASSERT_EQ(e1.j, 22);
   ASSERT_EQ(e1.b.s, "66");
 }
+
+TEST_F(OptionTypeInfoTest, TestVectorType) {
+  OptionTypeInfo vec_info = OptionTypeInfo::Vector<std::string>(
+      0, OptionVerificationType::kNormal, OptionTypeFlags::kNone, 0,
+      {0, OptionType::kString});
+  std::vector<std::string> vec1, vec2;
+  std::string mismatch;
+
+  ConfigOptions config_options;
+  TestAndCompareOption(config_options, vec_info, "v", "a:b:c:d", &vec1, &vec2);
```

Comment:
Better to add the test case for other possible input string format (like {A=a:a1:a2;B=b:b1}?)

Reply:
I am willing to add more test cases, but am not sure what exactly you are asking for.  Do you mean "A={a:a1:a2};" or A=a:{a1}:{a2}" or something else?   Your comment shows two separate options.

Reply:
Yeah, you are right. Sorry about the misleading in the example.

Reply:
Added more test cases for vectors with nested brackets and nested vectors